pymatgen/io/lammps/sets.py

```diff
@@ -0,0 +1,219 @@
+        keep_stages: Whether to keep the stage structure of the LammpsInputFile or not.
+
+    /!\ This InputSet and InputGenerator implementation is based on templates and is not intended to be very flexible.
+    For instance, pymatgen will not detect whether a given variable should be adapted based on others
+    (e.g., the number of steps from the temperature), it will not check for convergence nor will it actually run LAMMPS.
+    For additional flexibility and automation, use the atomate2-lammps implementation
+    (https://github.com/Matgenix/atomate2-lammps).
+    """
+
+    template: str = os.path.join(template_dir, "md.template")
+    calc_type: str = "lammps"
+
+    def __init__(
+        self, settings: dict[str, str | float] | None = None, keep_stages: bool = False  # pylint: disable=E1131
+    ):
+        self.settings = settings or {}
+        self.keep_stages = keep_stages
+
+        # load the template
+        with zopen(self.template, "r") as f:
+            template_str = f.read()
+
+        # replace all variables
+        self.input_str = Template(template_str).safe_substitute(**self.settings)
+        self.input_file = LammpsInputFile.from_string(self.input_str, keep_stages=self.keep_stages)
+
+    def get_input_set(  # type: ignore
+        self, structure: Structure | LammpsData | CombinedData | None  # pylint: disable=E1131
+    ) -> LammpsInputSet:
+        """
+        Generate a LammpsInputSet from the structure/data, tailored to the template file.
+        """
+        if isinstance(structure, Structure):
+            data = LammpsData.from_structure(structure)
+        else:
+            data = structure
+
+        # Get the LammpsInputSet from the input file and data
+        input_set = LammpsInputSet(
+            inputfile=self.input_file, data=data, calc_type=self.calc_type, template_file=self.template
+        )
+
+        return input_set
+
+
+class LammpsMinimization(BaseLammpsGenerator):
+    r"""
+    Yields a LammpsInputSet tailored for minimizing the energy of a system by iteratively
+    adjusting atom coordinates.
+    Example usage:
+    ```
+    structure = Structure.from_file("mp-149.cif")
+    lmp_minimization = LammpsMinimization(units="atomic").get_input_set(structure)
+    ```
+
+    Do not forget to specify the force field, otherwise LAMMPS will not be able to run!
+
+    /!\ This InputSet and InputGenerator implementation is based on templates and is not intended to be very flexible.
+    For instance, pymatgen will not detect whether a given variable should be adapted based on others
+    (e.g., the number of steps from the temperature), it will not check for convergence nor will it actually run LAMMPS.
+    For additional flexibility and automation, use the atomate2-lammps implementation
+    (https://github.com/Matgenix/atomate2-lammps).
+    """
+
+    template = os.path.join(template_dir, "minimization.template")
+    calc_type = "minimization"
+
+    def __init__(
```

Comment:
Please add a docstring describing the arguments here, as well as a type annotation so that users will understand they're going to a get a `LammpsInputSet`